src/Compilers/Core/Portable/InternalUtilities/ConcurrentDictionaryExtensions.cs

```diff
@@ -38,5 +38,25 @@ internal static class ConcurrentDictionaryExtensions
             return dictionary.GetOrAdd(key, boundFunction);
 #endif
         }
+
+        // original signature:
+        // public TValue ConcurrentDictionary<TKey, TValue>.AddOrUpdate<TArg>(TKey key, Func<TKey,TArg,TValue> addValueFactory, Func<TKey,TValue,TArg,TValue> updateValueFactory, TArg factoryArgument);
+        public static TValue AddOrUpdate<TKey, TValue, TArg>(
```

Comment:
>AddOrUpdate [](http://example.com/codeflow?start=29&length=11)

Is this extension method to provide the NETCOREAPP functionality to non-NETCOREAPP builds? If so, could the entire extension method be wrapped in `if !NETCOREAPP`?

Reply:
Other extensions in this file polyfill methods for netstandard2.0 in a similar way. I wanted to be consistent with them.